src/pydocstyle/config.py

```diff
@@ -195,7 +195,16 @@ def _get_config(self, node):
         if path in self._cache:
             return self._cache[path]
 
-        config_file = self._get_config_file_in_folder(path)
+        if self._options.config_location:
+            # Find the config directory
+            cli_config_dir = os.path.dirname(self._options.config_location)
+            if os.path.abspath(cli_config_dir) == os.getcwd():
+                config_file = self._options.config_location
+            else:
+                config_file = self._get_config_file_in_folder(path)
```

Comment:
I don't understand this logic. Why do you only allow configuration files from the current directory?